rust/headless-client/src/ipc.rs

```diff
@@ -1,11 +1,29 @@
+// There is no special way to prevent `cargo-mutants` from throwing false
+// positives on code for other platforms.
 #[cfg(target_os = "linux")]
-mod linux;
-#[cfg(target_os = "linux")]
-use linux as platform;
+#[path = "ipc/linux.rs"]
+pub mod platform;
 
 #[cfg(target_os = "windows")]
-pub mod windows;
-#[cfg(target_os = "windows")]
-pub use windows as platform;
+#[path = "ipc/windows.rs"]
+pub mod platform;
 
 pub(crate) use platform::{Server, Stream};
+
+/// A name that both the server and client can use to find each other
+#[derive(Clone, Copy)]
+pub enum ServiceId {
+    /// The IPC service used by Firezone GUI Client in production
+    ///
+    /// This must go in `/run/dev.firezone.client` on Linux, which requires
+    /// root permission
+    Prod,
+    /// An IPC service used for unit tests.
+    ///
+    /// This must go in `/run/user/$UID/dev.firezone.client` on Linux so
+    /// the unit tests won't need root.
+    ///
+    /// Includes an ID so that multiple tests can
+    /// run in parallel
+    Test(&'static str),
```

Comment:
If we keep the ctor that accepts a path, we don't need this right? We can just use a tempfile or something to have unique sockets per test.

Reply:
The trouble is, it's a `String` on Windows, not a `Path`, named pipes aren't part of the filesystem and they have odd restrictions on naming. So I don't think `tempfile` can generate a named pipe, and a UDS isn't a file anyway.

I made this enum to act as the abstraction over named pipes with `String`s and UDS with `PathBuf`s, and we don't want UDS' in the current working dir anyway, only under `/run/user/$UID` for unit tests and `/run/dev.firezone.client` for production systemd services

Reply:
Okay makes sense! Thanks for explaining. Perhaps we can have some docs that explain this design decision?